### PR TITLE
libgd: fix with_png on MSVC + modernize

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libgd/all**

building with `with_png=True` didn't work on MSVC

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
why didn't `with_png` work? I can't see a fix for that

**Comment by madebr** [comment]
> why didn't `with_png` work? I can't see a fix for that

`patches/2.3.x-png-msvc.patch` fixes it + added `getopt-for-visual-studio`

File: recipes/libgd/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
-project(libgd_test CXX)
+project(test_package C)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+conan_basic_setup(TARGETS)
 
-add_executable(${PROJECT_NAME} test_package.cpp)
-target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+find_package(libgd REQUIRED)
```
Comment (uilianries) [inline recipes/libgd/all/test_package/CMakeLists.txt:7]:
```suggestion
find_package(libgd CONFIG REQUIRED)
```

cmake_find_package_multi generates Cmake config file only

Comment (madebr) [reply recipes/libgd/all/test_package/CMakeLists.txt:7]: It doesn't really matter.
There is:
```cmake
find_package(libgd CONFIG REQUIRED)  # looks for libgd-config.cmake
find_package(libgd MODULE REQUIRED)  # looks for Findlibgd.cmake
find_package(libgd REQUIRED)  # looks for both Findlibgd.cmake and libgd-config
```

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**